tm-parsers/js/js.tm

```diff
@@ -1578,12 +1586,16 @@ TypeQuery -> TypeQuery :
     'typeof' TypeQueryExpression ;
 
 # 2.9
-ImportType -> ImportType :
-    'typeof'? 'import' '(' Type ')' ('.' TypeQueryExpression)? .noLineBreak TypeArguments? %prec resolveShift ;
+ImportType -> TsImportType:
+     ImportTypeStart .noLineBreak TypeArguments? %prec resolveShift ;
+
+ImportTypeStart -> TsImportTypeStart :
+  ('typeof'->TsTypeOf)? 'import' '(' Type ')'
+   | ImportTypeStart '.' IdentifierReference<~Yield, ~Await> -> TsImportTypeStart;
 
 TypeQueryExpression :
-    IdentifierNameRef
```

Comment:
I think this is also a recent fix - IdentifierNameRef is better/more permissive (allows keywords).
JS always allows keywords after a dot (see https://tc39.es/ecma262/#prod-CallExpression)